File: Level-114/Classwork/Classwork.js
// // ვესალმებით user-ს მოცემული გვაქვს user arr. ყველა user-ს ვესალმებით წინადადებით "helo{user}"" იმ შემთხვევაში თუ user აღმოჩნდება "Deme"

// const users = ["Luka", "Nika", "Mariami", "Deme", "Nia", "Cotne"]

// for (let i = 0; i < users.length; i++){
    
//     if (users[i] === "Deme"){
//         console.log("Baro Deme");
        
//         break
//     }
//     console.log(`Hello ${users[i]}`)
// }





// let arr = [1,2,3,4,5]

// // push - მასივის ბოლს ამატტებს ელემენტს 
// // pop - მასივის ბოლოდან აგდებს ელემენტს 
// // toString - მასივი გადაწავს String მონაცემის ტიპად 
// // at - მასივიდან აბრუნებს კონკრეტუილ ინდექსზე მდგომ ელემენტს, თუ დავტოვებტ ცარიელს აბრუნებს მენულე ინდექსზე მდგომ ელემენტს  
// // shift - აშორებს მენულზე ინდექსზე მყოფ ელემენტს 
// // unshift - ამატებს ელემენტებს სიის დასაწყისში



// arr.push(40)
// arr.pop()
// arr.shift()
// arr.unshift("apple", "strawberry", "blueberry")


// let num1 = arr.at(3)




// console.log(num1);


let languages = ["Python", "JS", "Java", "C++"];
// მოცემულია სიტყვების მასივი. იპოვე სიტყვა “Java”. როგორც კი იპოვი, დაბეჭდე და გააჩერე ლუპი.

for(let i = 0; i < languages.length; i++){
    if(languages[i] === "Java")
        console.log(languages[i]);
        break
}




// 2) დაბეჭდე რიცხვები 1-დან 150-მდე. როგორც კი შეხვდები რიცხვს, რომელიც იყოფა 6-ზე, მაგრამ არ იყოფა 12-ზე, გააჩერე ლუპი.

for (let i = 1;i < 150; i++){
    console.log(i);
    if(i % 6 == 0)
        break
}




// 3)შექმენით მასივი პროგრამირების ენებზე,შემდეგ push
// ის საშუალებით დაამატეთ ორი ელემენტი და შემდეგ პირველი და ბოლო ელემენტი ამოიღეთ შესაბამისი მეთოდებით,გამოიტანეთ მიღებული მასივი


let Proglanguages = ["Python", "Java", "JavaScript", "C", "C++", "PHP", "Go", "Rust", "Ruby",]


Proglanguages.push("Kotlin")
Proglanguages.pop()
Proglanguages.shift()

console.log(Proglanguages);

File: Level-113/Classwork/Classwork.py
# items = [
#     ("product1", 11),
#     ("product2", 9),
#     ("product3", 5),
#     ("product4", 13),
#     ("product5", 6),
#     ("product3", 23),
# ]

# # filter()

# # list => which obly contains products with price less than 10 

# filtered = list(filter(lambda item: item[1] <= 10, items))

# min_item = (filtered[0])[1]

# print(min_item)



# for i in filtered:
#     if i[1] < min_item:
#         min_item == i[1]
        
# print(f"The lowest price item: {min_item}")






names = ["Nika", "Luka", "Gvanca","Lazare","Deme"]


filtered_names = list(filter(lambda name: (name[0]).upper() == "L", names))


print(filtered_names)




# Probable classwork
# food = ["Salad H","Burger J","Fries J","Watermelon H", "Shawarma J","Carrot H", "Apple H"]




items = [
    ("product1", 11),
    ("product2", 9),
    ("product3", 5),
    ("product4", 13),
    ("product5", 6),
    ("product3", 23),
]

# map


# list only contains prices!

prices = list(map(lambda item: item[1], items ))



print(prices)


# Filter ის გამოყენებით:
# 1) languages = ["Python", "Java", "JavaScript", "C", "C++", "PHP", "Go", "Rust", "Ruby", "Kotlin"] დატოვე მხოლოდ ის ენები რომლების სიგრძეც მეტია 5ზე.

languages = ["Python", "Java", "JavaScript", "C", "C++", "PHP", "Go", "Rust", "Ruby", "Kotlin"]

filtered_langs = list(filter(lambda item: len.item > 5, languages))

print(filtered_langs)



# 2) foods = ["salad H", "burger J" , "fish H","fries J","fruits H","nuts H"]
# სიიდან დატოვე მხოლოდ ჯანსაღი საკვები —> H


foods = ["salad H", "burger J" , "fish H","fries J","fruits H","nuts H"]

filtered_foods = list(filter(lambda food: food[-1], foods))

print(filtered_foods)



# Map ის გამოყენებით:
# 3) languages = ["Python", "Java", "JavaScript", "C", "C++", "PHP", "Go", "Rust", "Ruby", "Kotlin"] თითოეულ ელემენტს წინ დაუმატეთ "I study"


languages = ["Python", "Java", "JavaScript", "C", "C++", "PHP", "Go", "Rust", "Ruby", "Kotlin"]




# 4) minutes = [60, 120, 180, 240, 30] მოცემულია წუთები სიაში,map() ის გამოყენებით წუთები გადაიყვანეთ საათებში და დაპრინტეთ
# Hint: hours = minutes / 60

minutes = [60, 120, 180, 240, 30]

mapped_minutes = list(map(lambda min: min(minutes / 60), minutes))

print(mapped_minutes)


